Classes/Http/Middleware/ProtectedResourceMiddleware.php

```diff
@@ -117,27 +116,23 @@ public function handle(ComponentContext $componentContext)
                 get_class($fileServeStrategy)), 1429704284);
         }
         /** @var ResponseInterface $httpResponse */
-        $httpResponse = $componentContext->getHttpResponse()
-            ->withHeader('Content-Type', $resource->getMediaType())
-            ->withHeader('Content-Disposition', 'attachment;filename="' . $resource->getFilename() . '"')
-            ->withHeader('Content-Length', $resource->getFileSize());
-
-        $this->emitResourceServed($resource, $httpRequest);
+        $httpResponse = new Response();
+        $httpResponse = $httpResponse->withHeader('Content-Type', $resource->getMediaType());
+        $httpResponse = $httpResponse->withHeader('Content-Disposition', 'attachment;filename="' . $resource->getFilename() . '"');
+        $httpResponse = $httpResponse->withHeader('Content-Length', $resource->getFileSize());
```

Comment:
I find that a little easier to comprehend:
```suggestion
        $httpResponse = new Response()
            ->withHeader('Content-Type', $resource->getMediaType());
            ->withHeader('Content-Disposition', 'attachment;filename="' . $resource->getFilename() . '"');
            ->withHeader('Content-Length', $resource->getFileSize());
```

Reply:
done post merge